devicetypes/axis/axis-gear-v5.1.src/axis-gear-v5.1.groovy

```diff
@@ -0,0 +1,420 @@
+    }
+    if (i > currentLevel) {
+        sendEvent(name: "windowShade", value: "opening")
+    } else if (i < currentLevel) {
+        sendEvent(name: "windowShade", value: "closing")
+    }
+    //setWindowShade(i)
+    if (state.currentVersion >= 1093){
+    	zigbee.command(CLUSTER_WINDOWCOVERING,WINDOWCOVERING_CMD_GOTOLIFTPERCENTAGE, zigbee.convertToHexString(100-i,2))
+    }
+    else {
+    	zigbee.setLevel(i)
+    }
+}
+
+//Send Command through setLevel()
+def open() {
+	log.info "open()"
+    sendEvent(name: "windowShade", value: "opening")
+    if (state.currentVersion >= 1093){
+		zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_OPEN)
+    }
+    else {
+    	setLevel(100)
+    }     
+}
+//Send Command through setLevel()
+def close() {
+	log.info "close()"
+	sendEvent(name: "windowShade", value: "closing")
+    if (state.currentVersion >= 1093){
+		zigbee.command(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_CMD_CLOSE)
+    }
+    else {
+    	setLevel(0)
+    }
+}
+
+def presetPosition() {
+	log.info "presetPosition()"
+    setLevel(preset ?: state.preset ?: 50)
+}
+
+//Reporting of Battery & position levels
+def ping(){
+	log.debug "Ping() "
+    //def cmds_ping = zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+    //return cmds_ping
+    return refresh()
+}
+
+//Set blind State based on position (which shows appropriate image) 
+def setWindowShade(value){
+ if ((value>0)&&(value<99)){
+    	sendEvent(name: "windowShade", value: "partially open", displayed:true)
+    } else if (value >= 99){
+    	sendEvent(name: "windowShade", value: "open", displayed:true)
+    }else{
+    	sendEvent(name: "windowShade", value: "closed", displayed:true)
+    }
+}
+
+//Refresh command
+def refresh() {
+	log.debug "parse() refresh"
+    def cmds_refresh = null
+    
+    if (state.currentVersion >= 1093){
+			cmds_refresh = zigbee.readAttribute(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE) +
+            				zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY) +
+                            zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+    }
+    else {
+			cmds_refresh = zigbee.readAttribute(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL) +
+            				zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY) +
+                            zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+    }
+    log.info "refresh() --- cmds: $cmds_refresh"
+
+	return cmds_refresh
+}
+
+def getversion () {
+	//state.currentVersion = 0
+    sendEvent(name: "version", value: "Checking Version ... ")     
+	return zigbee.readAttribute(CLUSTER_BASIC, BASIC_ATTR_SWBUILDID)
+}
+
+//configure reporting
+def configure() {   
+	state.currentVersion = 0
+	sendEvent(name: "windowShade", value: "unknown")
+    log.debug "Configuring Reporting and Bindings."
+    sendEvent(name: "checkInterval", value: 60 * 5, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```

Comment:
Generally when we set the check interval we like it to allow for the hub to have missed one periodic report + some fuzz time before marking the device offline. Right now you're configuring all your attributes to report once every hour, so unless you continually mess with the device or it has some other out-of-band reporting, you're going to be pinging this device every five minutes. I don't expect that would do great things for your battery life, and it doesn't do great things for anyone to rely on frequent pings to keep the device online. Based on your reporting intervals I would instead recommend a checkInterval time of 2 hours + 10 minutes `(2 * 60 * 60 + 10 * 60)`.